src/formats/chr.c

```diff
@@ -176,7 +191,12 @@ int sd_chr_load(sd_chr_file *chr, const char *filename) {
             chr->enemies[i]->pilot.total_value = trn.enemies[i]->total_value;
             chr->enemies[i]->pilot.photo_id = trn.enemies[i]->photo_id;
         }
-        memread_buf(mr, chr->enemies[i]->unknown, 25);
+        memread_buf(mr, chr->enemies[i]->unknown_a, 9);
+        chr->enemies[i]->photo_id = memread_ubyte(mr);
+        memread_buf(mr, chr->enemies[i]->unknown_b, 15);
+        if(trn_loaded) {
+            chr->enemies[i]->pilot.sex = pic.photos[(uint8_t)chr->enemies[i]->photo_id]->sex;
```

Comment:
is there a reason you're casting the uint16_t photo_id to a uint8_t?
if overreading the photos buffer is the concern, please perform a check against MAX_PIC_PHOTOS.

Reply:
This is a great question, I don't actually know the size of the field in the CHR file. That said, I don't think any PIC files with more than 255 entries exist..

Reply:
I made it a uint16_t now, better?

Reply:
Why are you casting?

Reply:
It was to fix a compiler warning.

Reply:
I ran a clang and gcc build (using CI, see PR #828) and do not see a compiler warning.

I suspect the warning was solved when you broke `char unknown[25]` into three: unknown_a, photo_id, and unknown_b-- and the cast is a vestige of the unknown char array's signedness.

Reply:
You seem to be correct.